src/jsutils/instanceOf.ts

```diff
@@ -1,10 +1,10 @@
 import { inspect } from './inspect.js';
 
 /* c8 ignore next 3 */
-const isProduction =
+const isDevelopment =
   globalThis.process != null &&
   // eslint-disable-next-line no-undef
-  process.env.NODE_ENV === 'production';
+  process.env.NODE_ENV === 'development';
```

Comment:
`test` and `development` should both invoke the non-production behavior.

Reply:
To minimize the size of this PR, I'd personally rewrite this to just:

```ts
const isProduction = process.env.NODE_ENV !== 'development' && process.env.NODE_ENV !== 'test';
```

Then the lower lines in this PR shouldn't be needed.

Reply:
965457044

so this is now:

```ts
const isProduction = globalThis.process == null || (process.env.NODE_ENV !== 'development' && process.env.NODE_ENV !== 'test');
```

retaining the check to see whether process is defined globally

Reply:
https://github.com/JoviDeCroock/graphql-minifier-experiments/pull/1

I have not gone through all the bundlers, but I think that flipping the default would require us changing our instructions about what to set globalThis.process to, now we would have to set to false

https://github.com/graphql/graphql-js/blob/16.x.x/website/docs/tutorials/going-to-production.md

and if I remember correctly, the reason that is in at all is to support cloudflare.

So I am not sure if I want to move forward with this PR actually.... as if we have a better long-term solution, we should move to that without this intermediate stage......?

Reply:
https://github.com/JoviDeCroock/graphql-minifier-experiments/pull/1

I have not gone through all the bundlers, but I think that flipping the default would require us changing our instructions about what to set globalThis.process to, now we would have to set to false

https://github.com/graphql/graphql-js/blob/16.x.x/website/docs/tutorials/going-to-production.md

and if I remember correctly, the reason that is in at all is to support cloudflare.

So I am not sure if I want to move forward with this PR actually.... as if we have a better long-term solution, we should move to that without this intermediate stage......?

@JoviDeCroock @benjie @phryneas

Reply:
Long-term solution: 3 versions of this file - one with the `development` export condition, without any of this logic, one with the `production` export condition without any of this, and one `default` that falls back to this (or a similar) solution.

Reply:
That does require us to start with an `exports` field though. #4096 might be a step in that direction - reviews welcome.

Reply:
Does an exports field help us for this internal function? Do we need an imports map as well?

Reply:
We would have two directions here:
* import map and hope that packagers actually support that nicely (I honestly do not know)
* exports field and we `import from 'graphql/instanceof'`

Reply:
It would be best not to export internal functions like instanceof at all. If bundlers respect import maps that would seem to be best.

Reply:
The problem is that e.g. Webpack 4 doesn't support import/export maps at all, but is afaik still used by tools like StoryBook.
Having `exports` in place that just also exist under the same path on the file system as a fallback can work in those cases - but `imports` maps unfortunately cannot :/

Reply:
Hmmm.

WebPack 5 was released on 2020.10.10 almost 4 years ago.

In terms of Storybook, which I have zero familiarity with, released v7.0.0 on 2023.04.12 almost 1.5 years ago and with that major [dropped support for Webpack 4.](https://storybook.js.org/docs/7/migration-guide).

Reply:
@markerikson I remember you had a very good reason to support Webpack 4 in the Redux packages. Is that obsolete with this?

Reply:
Submitted https://github.com/graphql/graphql-js/pull/4221 as an alternative demoing the use of an imports map.

Reply:
@phryneas I closed #4221 because once we use an imports map, it becomes tricky to use graphql in browsers in a bundler-free manner. I think unfortunately using the exports map and publishing instanceOf might have the same problem.

Reply:
@yaacovCR What problems are you seeing?
Generally, I'd say that unless we publish only a single file, users will have to adopt import maps, but that's the future of ESM on the web anyways.

Reply:
I’m not sure if we want to require the use of an import map but I would definitely defer to @JoviDeCroock  on that.